src/components/bookmark-button/index.tsx

```diff
@@ -1,35 +1,69 @@
-import { AUTH_ENABLED } from 'hooks/use-authentication'
+import { IconBookmarkAdd16 } from '@hashicorp/flight-icons/svg-react/bookmark-add-16'
+import { IconBookmarkRemove16 } from '@hashicorp/flight-icons/svg-react/bookmark-remove-16'
+import Button from 'components/button'
+import { withState } from './with-state'
 import { RemoveBookmarkIcon, AddBookmarkIcon } from './icons'
+import { BookmarkButtonConfigType, BookmarkButtonProps } from './types'
 import s from './bookmark-button.module.css'
 
+const bookmarkButtonConfig: BookmarkButtonConfigType = {
```

Comment:
Once I merge #839, there's room to consolidate this data with the toast messages also.